lib/purescript-cst/src/Language/PureScript/CST/Convert.hs

```diff
@@ -530,6 +531,57 @@ convertDeclaration fileName decl = case decl of
   ann =
     uncurry (sourceAnnCommented fileName) $ declRange decl
 
+  startSourcePos :: SourceToken -> Pos.SourcePos
+  startSourcePos = sourcePos . srcStart . tokRange . tokAnn
+
+  mkPartialInstanceName :: Maybe (Name Ident, SourceToken) -> QualifiedName (N.ProperName 'N.ClassName) -> [Type a] -> Either Text.Text N.Ident
+  mkPartialInstanceName nameSep cls args =
+    maybe (Left genName) (Right . ident . nameValue . fst) nameSep
+    where
+      -- truncate to 25 chars to reduce verbosity
+      -- of name and still keep it readable
+      -- unique identifier will be appended to this name
+      -- in desugaring proces
+      genName :: Text.Text
+      genName = Text.take 25 ("$" <> className <> "_" <> typeArgs) <> "$"
+
+      className :: Text.Text
+      className = do
+        let modName = (<> "_") . Text.replace "." "_" . N.runModuleName
+        (maybe "" modName (qualModule cls)) <> (N.runProperName $ qualName cls)
```

Comment:
I think this might be equivalent to `Text.replace "." "_" $ showQualified runProperName cls`.

Reply:
mm... doesn't seem like it. `showQualified` operates on `Language.PureScript.Names (Qualified)`, but `cls` is `Language.PureScript.CST.Types (QualifiedName)`

Reply:
Oh silly me, I didn’t pay attention to the types 🤦